crates/common/tedge_config/src/tedge_config_cli/tedge_config_dto.rs

```diff
@@ -52,68 +56,95 @@ pub(crate) struct TEdgeConfigDto {
 
 /// Represents the device specific configurations defined in the [device] section
 /// of the thin edge configuration TOML file
-#[derive(Debug, Default, Deserialize, Serialize)]
-pub(crate) struct DeviceConfigDto {
-    /// The unique id of the device (DEPRECATED)
-    /// This id is now derived from the device certificate
-    #[serde(rename(deserialize = "id"), skip_serializing)]
-    pub(crate) _id: Option<String>,
-
-    /// Path where the device's private key is stored.
-    /// Defaults to $HOME/.tedge/tedge-private.pem
-    pub(crate) key_path: Option<FilePath>,
-
-    /// Path where the device's certificate is stored.
-    /// Defaults to $HOME/.tedge/tedge-certificate.crt
-    pub(crate) cert_path: Option<FilePath>,
-
+#[derive(Debug, Default, Deserialize, Serialize, Document)]
+pub struct DeviceConfigDto {
+    /// Path where the device's private key is stored
+    #[doku(example = "/etc/tedge/device-certs/tedge-private-key.pem")]
+    #[doku(as = "PathBuf")]
+    pub(crate) key_path: Option<Utf8PathBuf>,
+
+    /// Path where the device's certificate is stored
+    #[doku(example = "/etc/tedge/device-certs/tedge-certificate.pem")]
+    #[doku(as = "PathBuf")]
+    pub(crate) cert_path: Option<Utf8PathBuf>,
+
+    /// The default device type
     #[serde(rename = "type")]
+    #[doku(example = "thin-edge.io")]
     pub(crate) device_type: Option<String>,
 }
 
+#[test]
+fn test_me_now() {
+    toml::from_str::<TEdgeConfigDto>(&doku::to_toml::<TEdgeConfigDto>()).unwrap();
+    println!("{}", doku::to_toml::<TEdgeConfigDto>());
+}
+
 /// Represents the Cumulocity specific configurations defined in the
 /// [c8y] section of the thin edge configuration TOML file
-#[derive(Debug, Default, Deserialize, Serialize)]
+#[derive(Debug, Default, Deserialize, Serialize, Document)]
 pub(crate) struct CumulocityConfigDto {
-    /// Preserves the current status of the connection
-    pub(crate) connect: Option<String>,
-
     /// Endpoint URL of the Cumulocity tenant
+    #[doku(example = "your-tenant.cumulocity.com")]
     pub(crate) url: Option<ConnectUrl>,
 
     /// The path where Cumulocity root certificate(s) are stored.
     /// The value can be a directory path as well as the path of the direct certificate file.
-    pub(crate) root_cert_path: Option<FilePath>,
-
-    /// Boolean whether Azure mapper adds timestamp or not.
-    pub(crate) mapper_timestamp: Option<bool>,
-
-    /// Set of c8y templates used for subscriptions.
+    #[doku(example = "/etc/tedge/c8y-trusted-root-certificates.pem")]
+    #[doku(as = "PathBuf")]
+    pub(crate) root_cert_path: Option<Utf8PathBuf>,
+
+    // TODO improve these examples
+    #[doku(example = "template1")]
+    #[doku(example = "template2")]
+    /// Set of c8y templates used for subscriptions
     pub(crate) smartrest_templates: Option<TemplatesSet>,
```

Comment:
Rather than the examples, I think what needs to be improved is the field documentation. These are not templates but template names and they must be defined on c8y.

Also, from the examples, it's not clear how one can set a *list* of template names. As far I understand `doku` these two examples are independent as those given for the `bind_address`. I would expect a sample that is actually a set.

Reply:
Yeah, the doku example is a bit weird, but this does translate to a list when generating the documentation.